### PR TITLE
Ab/view account2

### PR DESCRIPTION
**Pull request by AntoineBut**
The UI in ViewAccount should have been merged after this review : https://github.com/EPFLSWENT2024G1/partagix/pull/49#pullrequestreview-1985874067
I apologize for the particularly big PR, it should have been split in 2 after UI was done.

Old stuff : 
ViewAccount screen UI

New stuff : 
Database integration to read user data and fill information to the ViewAccount screen. 

App navigation to the account screen using the bottom NavBar

Note that even though this is working, the database is not currently storing the logged-in users. The default user displayed is actually in the database and not hard-coded.

### CODE REVIEW

File: app/src/main/java/com/android/partagix/model/UserViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          "User logged-in tried to watch current user profile but no user ID found. Defaulted to grinch user.")
+    }
+  }
+
+  private fun updateUIState(new: User) {
+    _uiState.value =
+        _uiState.value.copy(
+            user = new,
+        )
+    println("User updated: $new")
```
Comment (KassFlute) [inline app/src/main/java/com/android/partagix/model/UserViewModel.kt:67]: I wouldn't merge a println in the main (maybe commented)

Comment (KassFlute) [reply app/src/main/java/com/android/partagix/model/UserViewModel.kt:67]: (same for others println)

File: app/src/main/java/com/android/partagix/ui/App.kt [hunk 2]
```diff
@@ -124,8 +126,11 @@ class App(activity: MainActivity) : ComponentActivity(), SignInResultListener {
             inventoryViewModel = inventoryViewModel,
             navigateToTopLevelDestination = navigationActions::navigateTo)
       }
-      composable(Route.ACCOUNT) {
-        HomeScreen(navigationActions) /*TODO:Change to the account screen*/
+      composable(
+          Route.ACCOUNT,
+      ) {
+        println("navigated to account screen")
```
Comment (KassFlute) [inline app/src/main/java/com/android/partagix/ui/App.kt:132]: also a println

File: app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                    .testTag("mainContent")) {
+              /*LaunchedEffect(key1 = uiState) {
+                  println("refreshed User: $user")
+                  user = uiState.user
+
+              }*/
+              Image(
+                  painter =
+                      painterResource(
+                          id = R.drawable.ic_launcher_background) /*TODO: get profile picture*/,
```
Comment (KassFlute) [inline app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt:91]: "TODO: get profile picture" not part of this task ?

File: app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+                          unfocusedContainerColor = Color.Transparent,
+                          disabledContainerColor = Color.Transparent),
+                  modifier = Modifier.fillMaxWidth().padding(8.dp).testTag("rating"),
+                  readOnly = true,
+                  leadingIcon = { Icon(Icons.Default.CheckCircle, contentDescription = null) })
+              Spacer(modifier = Modifier.height(16.dp))
+              Row(modifier = Modifier.fillMaxWidth().padding(8.dp, 0.dp).testTag("actionButtons")) {
+                Button(
+                    onClick = { navigationActions.navigateTo(Route.INVENTORY) },
+                    modifier = Modifier.fillMaxWidth(0.5f).testTag("inventoryButton")) {
```
Comment (KassFlute) [inline app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt:171]: 0.5f could be a constant

File: app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+              Spacer(modifier = Modifier.height(8.dp))
+              Row(
+                  modifier = Modifier.fillMaxWidth().testTag("username"),
+                  horizontalArrangement = Arrangement.Absolute.SpaceAround) {
+                    val username = user.name
+                    Text("$username's profile")
+                  }
+              Spacer(modifier = Modifier.height(16.dp))
+
+              TextField(
```
Comment (KassFlute) [inline app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt:104]: We can select (highlight it by clicking on it) this field making the user think it is editable -> maybe it should be not selectable.

File: app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+                      }
+                      5 -> {
+                        "★★★★★ ($rating/5)"
+                      }
+                      else -> {
+                        "..."
+                      }
+                    }
+              }
+              TextField(
```
Comment (KassFlute) [inline app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt:152]: Same remark as the location field above

**Review by KassFlute — CHANGES_REQUESTED** [review]
Except a few little details (listed in code comment) it's all good.
-Maybe it's my emulator but I have the feeling that the user information are present shortly after loading the screen then disappear then appears again -> If it's really doing this, it should be addressed
-Maybe a bit more comments would be welcome

**Comment by KassFlute** [comment]
SonarCloud still can't get line coverage but it's not a problem from this

File: app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt (hunk 3 above)
Comment (AntoineBut) [reply app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt:91]: No, the database does not allow to store images yet

File: app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt (hunk 5 above)
Comment (AntoineBut) [reply app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt:104]: This has been addressed

File: app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt (hunk 4 above)
Comment (AntoineBut) [reply app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt:171]: Changed the way the UI displays the 2 buttons to be more simple.

File: app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt (hunk 5 above)
Comment (KassFlute) [reply app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt:98]: nice

File: app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt (hunk 4 above)
Comment (KassFlute) [reply app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt]: makes more sense like this !

**Comment by KassFlute** [comment]
"Maybe it's my emulator but I have the feeling that the user information are present shortly after loading the screen then disappear then appears again -> If it's really doing this, it should be addressed"
-> turns out to be a problem with the navigation bar so not this pr.

Everything is good for me.

**Pull request merged**